src/main/java/org/skife/jdbi/v2/unstable/Objects.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.skife.jdbi.v2.unstable;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+// TODO merge back into BindIn?
+final class Objects {
+    private Objects() {}
+
+    static Object[] toArray(final Object obj) {
+        if (obj instanceof Iterable) {
+            final Iterable<?> iterable = (Iterable) obj;
+            final List<Object> out = new ArrayList<Object>();
+
+            for (final Object x : iterable) {
+                out.add(x);
+            }
+
+            return out.toArray();
+        }
+
+        if (obj.getClass().isArray()) {
+            if (obj instanceof Object[]) {
+                return (Object[]) obj;
+            } else {
+                final int length = Array.getLength(obj);
+                final Object[] out = new Object[length];
+
+                for (int i = 0; i < length; i++) {
+                    out[i] = Array.get(obj, i);
+                }
+
+                return out;
+            }
+        }
+
+        throw new IllegalArgumentException("argument must be one of the following: Iterable, or an array/varargs (primitive or complex type)");
+    }
+
+    static int arraySize(final Object obj) {
+        if (obj instanceof Iterable) {
```

Comment:
It may be worthwhile to check for `Collection` first, and return `((Collection)obj).size()`.